gfosd/components/base_graph_class.py

```diff
@@ -26,6 +26,7 @@ class GraphComponent(ABC):
     def __init__(self, weight=1, diff=0, **kwargs):
         self._weight = weight
         self._diff = diff
+        self._has_helpers = diff != 0
```

Comment:
Added the `_has_helpers` attribute to `GraphComponent` as I figured this is an easy way to keep track of which components have helper variables. 

My thought is that when we implement components that have helper variables, we simply set this flag to `True`, as I do in the `Basis` class.